gammapy/maps/core.py

```diff
@@ -1907,3 +1907,76 @@ def sample_coord(self, n_events, random_state=0):
         cdict = OrderedDict(zip(self.geom.axes_names, coords))
 
         return MapCoord.create(cdict, frame=self.geom.frame)
+
+    def dot(self, other):
```

Comment:
This should probably be `__matmul__`.

Reply:
Or probably both. There might be cases where we need an additional `.dot(other, common_axis_name="")` to specify which axis to sum over. If it is ambiguous it is currently just implemented to fail, which is fine. But we might change that in future...

Reply:
But I'm definitely +1 to support the `@` operator...!